UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.CleanUpJob/Helpers/AzureFileSystemHelper.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.Extensions;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Logging;
+
+namespace UKHO.ExchangeSetService.CleanUpJob.Helpers
+{
+    [ExcludeFromCodeCoverage]
+    public class AzureFileSystemHelper : IAzureFileSystemHelper
+    {
+        private readonly IAzureBlobStorageClient azureBlobStorageClient;
+        private readonly ILogger<AzureFileSystemHelper> logger;
+
+        public AzureFileSystemHelper(IAzureBlobStorageClient azureBlobStorageClient,
+                                    ILogger<AzureFileSystemHelper> logger)
+        {
+            this.azureBlobStorageClient = azureBlobStorageClient;
+            this.logger = logger;
+        }
+
+        public async Task<bool> DeleteDirectoryAsync(int numberOfDays, string storageAccountConnectionString, string containerName, string filePath)
+        {
+            bool deteleFilesStatus = false;
+            try
+            {
+                var subFolder = Directory.GetDirectories(filePath);
+                var historicDateTimeInUtc = DateTime.UtcNow.AddDays(-numberOfDays);
+                DateTime historicDate = new DateTime(historicDateTimeInUtc.Year, historicDateTimeInUtc.Month, historicDateTimeInUtc.Day);
+
+                foreach (var subFolderName in subFolder)
+                {
+                    DateTime subFolderDateTime;
+                    string dateFolder = new DirectoryInfo(subFolderName).Name;
+                    bool isValidDate = DateTimeExtensions.IsValidDate(dateFolder, out subFolderDateTime);
+                    
+                    if (isValidDate)
+                    {
+                        DateTime subFolderDate = new DateTime(subFolderDateTime.Year, subFolderDateTime.Month, subFolderDateTime.Day);
+                        
+                        if (subFolderDate <= historicDate)
+                        {
+                            DirectoryInfo di = new DirectoryInfo(subFolderName);
+
+                            foreach (DirectoryInfo dir in di.GetDirectories())
+                            {
+                                var batchId = dir.Name;
+                                string scsResponseFileName = new DirectoryInfo(batchId).Name + ".json";
+                                var response = await azureBlobStorageClient.DeleteFileFromContainer(storageAccountConnectionString, containerName, scsResponseFileName);
+                                if (response)
+                                {
+                                    dir.Delete(true);
+                                    logger.LogInformation(EventIds.HistoricDateFolderDeleted.ToEventId(), "SCS response json file {ScsResponseFileName} deleted successfully from the container.", scsResponseFileName);
```

Comment:
Kindly change eventid to HistoricSCSResponseFileDeleted

Reply:
i have done changes eventid to HistoricSCSResponseFileDeleted.